react-backend/client/src/components/admin/adminCBSettingsPage.js

```diff
@@ -135,7 +138,36 @@ export default class AdminCBSettingsPage extends Component {
       email,
       uploadedFileCloudinaryUrl,
     })
-      .then(res => res.details)
+      .then((res) => {
+        if (res.status === 400) {
+          return res.text();
+        }
+        return res;
+      })
+      .then((data) => {
+        console.log(data);
+
+        switch (data) {
+          case 'email':
+            this.setError([errorMessages.EMAIL_ERROR]);
+            break;
+          case 'name':
+            this.setError([errorMessages.NAME_ERROR]);
+            break;
+          case 'emailname':
+            this.setError([
+              errorMessages.NAME_ERROR,
+              errorMessages.EMAIL_ERROR,
+            ]);
+            break;
+          case 'noinput':
+            this.setError([errorMessages.NO_INPUT_ERROR]);
+            break;
+          default:
+            return data.details;
```

Comment:
this is redundant if you've got `return` on line 169, you can just put `break` here